api/cpp/esp-idf/slint/cmake/FindSlint.cmake

```diff
@@ -0,0 +1,87 @@
+#    and make it available.
+#
+# The following variables may be set to affect the behaviour:
+#
+# `SLINT_TARGET_ARCHITECTURE`: Set this to the desired target architecture. The format of this string is matched against
+# the `Slint-cpp-*-$SLINT_TARGET_ARCHITECTURE.tar.gz` pre-built assets on the GitHub releases. For example, if you're targeting
+# STM32 ARM architectures, you'd set this to `thumbv7em-none-eabihf`. If not set, this module will attempt to detect if compilation
+# is happening in an ESP-IDF cross-compilation environment and detect the architecture accordingly, otherwise
+# `${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}` is used.
+#
+# `SLINT_USE_NIGHTLY_VERSION`: When `find_package(Slint)` is called with a version, then this module will 
+# attempt to download a pre-compiled binary from the nightly snapshot of Slint.
+
+find_package(Slint ${Slint_FIND_VERSION} QUIET CONFIG)
+if (TARGET Slint::Slint)
+    return()
+endif()
+
+if (NOT SLINT_TARGET_ARCHITECTURE)
+    if(WIN32)
+        if(MSVC)
+            set(compiler_suffix "-MSVC")
+        elseif(MINGW)
+            set(compiler_suffix "-MinGW")
+        endif()
+        if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+            set(CPACK_SYSTEM_NAME win64)
+        else()
+            set(CPACK_SYSTEM_NAME win32)
+        endif()
+        set(SLINT_TARGET_ARCHITECTURE "${CPACK_SYSTEM_NAME}${compiler_suffix}")
+    elseif (CONFIG_IDF_TARGET_ARCH_XTENSA)
+        set(SLINT_TARGET_ARCHITECTURE "xtensa-${IDF_TARGET}-none-elf")
+    elseif(CONFIG_IDF_TARGET_ARCH_RISCV)
+        if (CONFIG_IDF_TARGET_ESP32C6 OR CONFIG_IDF_TARGET_ESP32C5 OR CONFIG_IDF_TARGET_ESP32H2)
+            set(SLINT_TARGET_ARCHITECTURE "riscv32imac-esp-espidf")
+        elseif (CONFIG_IDF_TARGET_ESP32P4)
+            set(SLINT_TARGET_ARCHITECTURE "riscv32imafc-esp-espidf")
+        else ()
+            set(SLINT_TARGET_ARCHITECTURE "riscv32imc-esp-espidf")
+        endif()
+    else()
+        set(SLINT_TARGET_ARCHITECTURE "${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}")
+    endif()
+endif()
+
+if (NOT DEFINED Slint_FIND_VERSION)
+    return()
+endif()
+
+set(slint_version "${Slint_FIND_VERSION}")
+set(github_release "${slint_version}")
+
+if (SLINT_USE_NIGHTLY_VERSION)
+    set(github_release "nightly")
+    # Set this to instruct the slint-compiler download to use the same release
+    set(SLINT_GITHUB_RELEASE "nightly" CACHE STRING "")
+endif()
+
+set(prebuilt_archive_filename "Slint-cpp-${slint_version}-${SLINT_TARGET_ARCHITECTURE}.tar.gz")
+set(download_target_path "${CMAKE_BINARY_DIR}/slint-prebuilt/")
+set(download_url "https://github.com/slint-ui/slint/releases/download/${github_release}/${prebuilt_archive_filename}")
+
+file(MAKE_DIRECTORY "${download_target_path}")
+message(STATUS "Downloading pre-built Slint binary ${download_url}")
+file(DOWNLOAD "${download_url}" "${download_target_path}/${prebuilt_archive_filename}" STATUS download_status)
```

Comment:
Does `file(DOWNLOAD` cache the result? if no, we should do something about that. If yes, will the message not be misleading if it was already downloaded?

Reply:
> Or would you prefer just to not download if the file exists?

Yes, I think we should do that. But can be done in a followup

Reply:
Ok, will do tomorrow.